game-core/src/main/java/games/strategy/engine/framework/startup/login/ClientLoginValidator.java

```diff
@@ -92,8 +91,6 @@ public String verifyConnection(
 
     if (hashedMac == null) {
       return ErrorMessages.UNABLE_TO_OBTAIN_MAC;
-    } else if (!MacFinder.isValidHashedMacAddress(hashedMac)) {
```

Comment:
System id does not have the same hashing applied to it, as it's random. Since the system id is client-side generated, any validity requirements based on hashing can be elided. Hence, we're making it more difficult for system-id/mac to be spoofed, but ultimately we're not really adding security.

I'm curious if we think we should try to make it a bit harder for someone. The scenario is basically someone changes their system id by custom compiling code, do we think that a hash guarantee is going to actually stop them if they get an error for invalid system id, or will the attacker be sophisticated enough already to hash their injected value so it passes?

Reply:
I think we can all agree here that the system can be tricked with ease if someone actually tried, so I wouldn't bother too much with the details.
The only people getting to this point, won't get tricked by a simple additionall step